sdk/src/main/java/com/mapbox/maps/ViewAnnotationManagerImpl.kt

```diff
@@ -308,23 +309,80 @@ internal class ViewAnnotationManagerImpl(
     return null to null
   }
 
+  private class PositionDiffCallback(
+    val oldList: List<ViewAnnotationPositionDescriptor>,
+    val newList: List<ViewAnnotationPositionDescriptor>,
+  ) : DiffUtil.Callback() {
+    override fun getOldListSize() = oldList.size
+    override fun getNewListSize() = newList.size
+    override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int) = oldList[oldItemPosition].identifier == newList[newItemPosition].identifier
+    override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int) = oldList[oldItemPosition].identifier == newList[newItemPosition].identifier
+  }
+
+  private var positionDiffCallback = PositionDiffCallback(emptyList(), emptyList())
+
   private fun positionAnnotationViews(
     positionDescriptorCoreList: List<ViewAnnotationPositionDescriptor>
   ) {
+    val positionDiffCallback = PositionDiffCallback(
+      oldList = positionDiffCallback.newList,
+      newList = positionDescriptorCoreList
+    )
+
+    var needToReorderZ = false
+    DiffUtil.calculateDiff(positionDiffCallback).dispatchUpdatesTo(
```

Comment:
wdyt about async diff calculation? 
maybe if you have 1000+ elements there might be some performance issues